src/main/kotlin/app/utils/logging/LoggerUtils.kt

```diff
@@ -0,0 +1,227 @@
+package app.utils.logging
+
+/**
+This file had simple json logging utils extending SLF4J in a standard way.
+It brings all our changes into one code module and one test file, and allows us to make a simple override in the
+resources/logback.xml file so that all vagaries are in code and unit-testable; one "just" does this:
+
+<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+<encoder class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
+<layout class="app.utils.logging.LoggerLayoutAppender" />
+</encoder>
+</appender>
+
+See http://logback.qos.ch/manual/layouts.html for examples.
+ */
+
+import app.batch.getEnv
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.spi.ThrowableProxyUtil
+import ch.qos.logback.core.CoreConstants
+import ch.qos.logback.core.LayoutBase
+import org.apache.commons.text.StringEscapeUtils
+import org.slf4j.Logger
+import java.net.InetAddress
+import java.text.SimpleDateFormat
+import java.util.*
+
+private val UNSET_TEXT = "NOT_SET"
+private val defaultFormat = makeUtcDateFormat() // 2001-07-04T12:08:56.235
+
+private var hostname = InetAddress.getLocalHost().hostName
+private var environment = System.getProperty("environment", UNSET_TEXT)
+private var application = System.getProperty("application", UNSET_TEXT)
+private var app_version = System.getProperty("app_version", UNSET_TEXT)
+private var component = System.getProperty("component", UNSET_TEXT)
+private var correlation_id = System.getProperty("correlation_id", UNSET_TEXT)
+private var staticData = makeLoggerStaticDataTuples()
+
+val data_table = getEnv("K2HB_HBASE_DATA_TABLE") ?: "k2hb:ingest"
```

Comment:
Do these really belong here - i.e. are they more than just logging config?

Reply:
Well - I want them logged as fist class citizens, and they must also be in the hbase config. Both are equally important - it's not "just logs", but a key part of or monitoring output... Could go either way around...